frontend/public/components/instantiate-template.tsx

```diff
@@ -147,7 +147,7 @@ class TemplateForm_ extends React.Component<
   }
 
   componentDidUpdate(prevProps: TemplateFormProps & WithTranslation) {
-    if (this.props.obj !== prevProps.obj) {
+    if (this.props.obj.data.parameters !== prevProps.obj.data.parameters) {
```

Comment:
Looks good. Just one more tiny suggestion; Data might not be loaded immediately so it would be better to have null check
```suggestion
    if (this.props.obj.data?.parameters !== prevProps.obj.data?.parameters) {
```